airflow/models/baseoperator.py

```diff
@@ -292,6 +311,12 @@ class derived from this one results in the creation of a task object,
     # Defines if the operator supports lineage without manual definitions
     supports_lineage = False
 
+    # If True then the class constructor was called
+    _instantiated = False
```

Comment:
```suggestion
    __instantiated = False
```

I think we can do this, which then means this is much much less likely to clash with any fields in a subclass.

Internally Python converts `__x` to `_BaseOperator__x` when used: https://docs.python.org/3/tutorial/classes.html#tut-private